### PR TITLE
Add optional property combinator

### LINKED ISSUE #29: Add support for optional record properties
I have a rough thing that works for this using an `# optionalProp` combinator rather than the record syntactic sugar, but would like to come up with a way that can work for sugar too.

### PR DESCRIPTION
**Pull request by garyb**
Partially resolves #29 - no record-based sugar for it still, but it's still useful in this form.

/cc @gabejohnson @kritzcreek

### CODE REVIEW

**Comment by wclr** [comment]
That would be really cool. Is it going to land?

File: src/Data/Codec/Argonaut/Compat.purs [hunk 1]
```diff
@@ -2,15 +2,15 @@
 module Data.Codec.Argonaut.Compat
   ( module Data.Codec.Argonaut.Compat
   , module Data.Codec.Argonaut
-  , module Common
+  , module Common
```
Comment (srghma) [inline src/Data/Codec/Argonaut/Compat.purs:5]: trailing space

File: test/Test/Prim.purs [hunk 2]
```diff
@@ -55,32 +58,41 @@ main = do
...
+propString = propCodec genAsciiString CA.string
 
 propChar ∷ Gen Result
-propChar = propCodec genAsciiChar JA.char
+propChar = propCodec genAsciiChar CA.char
 
-propJArray ∷ Gen Result
-propJArray = propCodec'' (show <<< map J.stringify) (Gen.unfoldable genJson) JA.jarray
+propCArray ∷ Gen Result
+propCArray = propCodec'' (show <<< map J.stringify) (Gen.unfoldable genJson) CA.jarray
```
Comment (srghma) [inline test/Test/Prim.purs:92]: should be called propJArray

File: test/Test/Prim.purs [hunk 3]
```diff
@@ -8,18 +8,21 @@ import Data.Argonaut.Core as J
 import Data.Argonaut.Gen (genJson)
 import Data.Char.Gen (genAsciiChar)
 import Data.Codec.Argonaut.Common ((~))
-import Data.Codec.Argonaut.Common as JA
+import Data.Codec.Argonaut.Common as CA
```
Comment (srghma) [inline test/Test/Prim.purs:11]: even better name is the full name - `Codec.Argonaut` or `CodecArgonaut`

**Comment by thomashoneyman** [comment]
This was done in #49

**Pull request closed without merging**